app/models/agents/http_status_agent.rb

```diff
@@ -60,11 +61,12 @@ def receive(incoming_events)
     private
 
     def check_this_url(url)
-      if result = ping(url)
-        create_event payload: { 'url' => url, 'status' => result.status.to_s, 'response_received' => true }
-        memory['last_status'] = result.status.to_s
+      measured_result = TimeTracker.track { ping(url) }
```

Comment:
`TimeTracker` seems useful.  I'm curious what the advantages over something like this are, though:

```ruby
start = Time.now.to_f
result = ping(url)
total_time = Time.now.to_f - start
if result
  create_event payload: { 'url' => url, 'status' => measured_result.status.to_s, 'response_received' => true, 'elapsed_time' => total_time }
...
```